apps/condo/domains/news/schema/ExportNewsRecipientsService.js

```diff
@@ -0,0 +1,242 @@
+            },
+        },
+    })
+
+    return linkToFile
+}
+
+/**
+ * List of possible errors, that this custom schema can throw
+ * They will be rendered in documentation section in GraphiQL for this custom schema
+ */
+const errors = {
+    MISSING_PROPERTY: {
+        mutation: 'exportNewsRecipients',
+        code: INTERNAL_ERROR,
+        type: NOT_FOUND,
+        message: 'Could not found property related to newsItemScope',
+        messageForUser: 'api.user.exportNewsRecipients.MISSING_PROPERTY',
+    },
+    WRONG_ORGANIZATION_ID: {
+        mutation: 'exportNewsRecipients',
+        code: INTERNAL_ERROR,
+        type: NOT_FOUND,
+        message: 'Could not found organization by provided id',
+        messageForUser: 'api.user.exportNewsRecipients.WRONG_ORGANIZATION_ID',
+    },
+}
+
+const ExportNewsRecipientsService = new GQLCustomSchema('ExportNewsRecipientsService', {
+    types: [
+        {
+            access: true,
+            type: 'input ExportNewsRecipientsInput { dv: Int!, sender: JSON!, organizationId: ID!, newsItemScopes: [NewsItemScopeWhereInput] }',
+        },
+        {
+            access: true,
+            type: 'type ExportNewsRecipientsOutput { status: String!, linkToFile: String! }',
+        },
+    ],
+    
+    mutations: [
+        {
+            access: access.canExportNewsRecipients,
+            schema: 'exportNewsRecipients(data: ExportNewsRecipientsInput!): ExportNewsRecipientsOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { data: { newsItemScopes, organizationId } } = args
+                const locale = extractReqLocale(context.req) || conf.DEFAULT_LOCALE
+
+                const organization = await Organization.getOne(context, {
+                    id: organizationId,
+                    deletedAt: null,
+                })
+                if (!organization) throw new GQLError(errors.WRONG_ORGANIZATION_ID, context)
+
+                let properties
+                if (!isEmpty(newsItemScopes)) {
+                    properties = newsItemScopes.map(newsItemScope => {
+                        if (!isNil(newsItemScope.property)) return newsItemScope.property.id
+                    })
+                }
+
+
+                let residentsByProperties
+                if (!isEmpty(properties)) {
+                    residentsByProperties = await loadListByChunks({
+                        context: context,
+                        list: Resident,
+                        chunkSize: 50,
+                        limit: 100000,
+                        where: {
+                            property: {
+                                id_in: properties,
+                            },
+                            deletedAt: null,
+                        },
+                    })
+                }
+
+                const recipientsByOrganization = []
+                if (organizationId && isEmpty(newsItemScopes)) {
```

Comment:
Looks like `organizationId` is mandatory according to the input type. So, there is no need to check it. Anyway, you didn't check it when load the organization model.

Reply:
Agree. I wanted a check that would catch the case when the deleted organization is passed. But it seems that this is not necessary, since this service is used only on the client, and there is no access to deleted entity instances